src/inc/quic_platform.h

```diff
@@ -67,16 +67,16 @@ typedef struct QUIC_SINGLE_LIST_ENTRY {
 // Different pool tags used for marking allocations.
 //
 
-#define QUIC_POOL_GENERIC   'CIUQ'  // QUIC - Generic QUIC
-#define QUIC_POOL_CONN      'noCQ'  // QCon - QUIC connection
-#define QUIC_POOL_TP        'PTCQ'  // QCTP - QUIC connection transport parameters
-#define QUIC_POOL_STREAM    'mtSQ'  // QStm - QUIC stream
-#define QUIC_POOL_SBUF      'fBSQ'  // QSBf - QUIC stream buffer
-#define QUIC_POOL_META      'MFSQ'  // QSFM - QUIC sent frame metedata
-#define QUIC_POOL_DATA      'atDQ'  // QDta - QUIC datagram buffer
-#define QUIC_POOL_TEST      'tsTQ'  // QTst - QUIC test code
-#define QUIC_POOL_PERF      'frPQ'  // QPrf - QUIC perf code
-#define QUIC_POOL_TOOL      'loTQ'  // QTol - QUIC tool code
+#define QUIC_POOL_GENERIC   *(uint32_t*)"CIUQ"  // QUIC - Generic QUIC
```

Comment:
These worry me. The generated assembly is not the same, at least without optimization. 

`mov     edi, 1128879441` for the multichar

```
.LC0:
        .string "noCQ"

        mov     eax, OFFSET FLAT:.LC0
        mov     eax, DWORD PTR [rax]
        mov     edi, eax
```

I know that they realistically are the same, but I think I would prefer the warning suppression and identical assembly between debug and release. @nibanks what do you think?

Reply:
I tend to agree.

Reply:
OR, we can have a `QUIC_TAG` macro that abstracts difference between platforms here.

Reply:
You could define them as hex constants instead of strings...

Reply:
Linux was where I was seeing the assembly difference. I don't think either one is standard, but I'd prefer the multichar over type punning a string, as its implementation defined instead of potentially being UB.

Reply:
I prefer to keep Windows as it is, as this is the standard mechanism for how Windows (internally) defines pool tags.

Reply:
E.g., 0x74735451 instead of 'tsTQ'

Reply:
That'd I'd be more fine with. But keep windows the multichars. I like the macro idea.

Reply:
I implemented the macro idea and left the multichar for Windows (which makes the non-Windows macro quite ugly...)